OtripleS.Web.Api.Tests.Unit/Services/StudentServiceTests/StudentServiceTests.cs

```diff
@@ -36,5 +38,35 @@ private Student CreateRandomStudent()
 
             return filler.Create();
         }
+
+        private StudentUpdateDto CreateRandomDto()
+        {
+            var filler = new Filler<StudentUpdateDto>();
+            filler.Setup()
+                .OnProperty(dto => dto.BirthDate).Use(this.dateTimeBroker.GetCurrentDateTime().AddYears(-21));
+            return filler.Create();
+        }
+
+        private Student NewStudentWithUpdatedProperties(Student student, StudentUpdateDto dto)
+        {
+            var newStudent = new Student
+            {
+                IdentityNumber = dto.IdentityNumber,
+                Gender = GenderHelper.StringToGenderConverter(dto.Gender),
+                BirthDate = dto.BirthDate,
+                FirstName = dto.FirstName,
+                MiddleName = dto.MiddleName,
+                LastName = dto.LastName,
+                Id = student.Id,
+                UserId = student.UserId,
+                CreatedBy = student.CreatedBy,
+                CreatedDate = student.CreatedDate,
+                UpdatedBy = student.UpdatedBy,
+                UpdatedDate = student.UpdatedDate
+            };
```

Comment:
I don't think we need a DTO for that, we just need to get an object from the API request and validate the following:
0. Whether the object id is referencing a real id
1. CreatedDate matches the storage 
2. Updated date is ahead of updated date in storage and not in the future
3. The rest of the structural validations we do for add, like name should not be empty or numbers ...etc.

Reply:
//given
       ```
            Student randomStudent = CreateRandomStudent();
            Student inputStudent = randomStudent;
            Student storageStudent = inputStudent;
            Student expectedStudent = storageStudent;
```
       with this setup, all 4 of them are the referencing same object, so if changes are made to storage student it will always be the same as expected student.

should I change this test?

Reply:
I have made all the other changes as you asked. but i could not find a way to pass the 2. The updated date is ahead of updated date in storage

Reply:
Please take a look if the changes satisfy your requirements

Reply:
@hassanhabib can you please help e how i should test the updated date is after the storageStudent? since i am returning the storageStudent in mock it will always be the same.

Reply:
You need to clone the storage student, and modify the updated date on the clone.
There's a library called `DeepCloner` pull it in and clone storage then modify the data on the `inputStudent`.